tests/serac_sets.cpp

```diff
@@ -0,0 +1,296 @@
+    auto stripe_height = flag_height / 13;
+    auto stripe        = static_cast<int>(coords[1] / stripe_height);
+    return stripe % 2 ? 2. : 1.;
+  });
+
+  // blue portion
+  // blue = 3, white 2, outside = 1
+  auto stars_coef = mfem::FunctionCoefficient([=](const mfem::Vector& coords) {
+    auto   blue_height = flag_height / 2;
+    auto   blue_width  = flag_width / 2.5;
+    double color       = 1.;  // not blue
+    if (coords[0] <= blue_width && coords[1] >= blue_height) {
+      // could be blue, check to see if it's in a "star"
+      // 9 down 11 across
+      auto checker_height = blue_height / 9;
+      auto checker_width  = blue_width / 11;
+      auto checker_x      = static_cast<int>(coords[0] / checker_width);
+      auto checker_y      = static_cast<int>((coords[1] - blue_height) / checker_height);
+      if ((checker_x + checker_y * 11) % 2 == 0) {
+        color = 2;
+      } else {
+        color = 3;
+      }
+    }
+    return color;
+  });
+
+  mfem::VisItDataCollection visit(pmesh->GetComm(), "flag", pmesh.get());
+
+  serac::FiniteElementState stripes(
+      *pmesh, FiniteElementState::Options{.order      = 0,
+                                          .vector_dim = 1,
+                                          .coll = {std::make_unique<mfem::L2_FECollection>(0, pmesh->SpaceDimension())},
+                                          .ordering = mfem::Ordering::byVDIM,
+                                          .name     = "stripes",
+                                          .alloc_gf = true});
+  stripes.project(stripes_coef);
+
+  auto stripes_attr_set =
+      Set(mfem_ext::MakeAttributeList<std::vector<int>>(*pmesh, stripes_coef, mfem_ext::digitize::floor));
+
+  serac::FiniteElementState stars(
+      *pmesh, FiniteElementState::Options{.order      = 0,
+                                          .vector_dim = 1,
+                                          .coll = {std::make_unique<mfem::L2_FECollection>(0, pmesh->SpaceDimension())},
+                                          .ordering = mfem::Ordering::byNODES,
+                                          .name     = "stars",
+                                          .alloc_gf = true});
+  stars.project(stars_coef);
+
+  auto stars_attr_set =
+      Set(mfem_ext::MakeAttributeList<std::vector<int>>(*pmesh, stars_coef, mfem_ext::digitize::floor));
+
+  // Set two additional bdr attributes: 1 = everything, 2 = bottom, 3 = right side
+  auto bdr_assign_coef = mfem::FunctionCoefficient([&](const mfem::Vector& x) {
+    if (fabs(x[1] - 0.) < 1.e-5) {
+      return 2.;
+    } else if (fabs(x[0] - flag_width) < 1.e-5) {
+      return 3.;
+    }
+    return 1.;
+  });
+
+  mfem_ext::AssignMeshBdrAttributes(
+      *pmesh, mfem_ext::MakeBdrAttributeList<std::vector<int>>(*pmesh, bdr_assign_coef, mfem_ext::digitize::floor));
+
+  // we'll get the top blue corner
+  auto blue = stars_attr_set.getComplement({1});
```

Comment:
This method finds all the index - attribute pairs that are the complement of `1` to get the "blue/stars" component of domain defined by `stars_coef` (only element indices with attributes 2,3). 

The `stripes_attr_set` contains the entire domain where element index attributes of 1 ("red") and 2 ("white") are assigned. `getDifference` provides the set difference of the entire domain and the blue portion of the flag and yields only the ("red/white") element indices that do not overlap with the top left blue portion.

Lastly we combine the stars with the stripes to get a domain which has elements that look like a flag (red = 1, white = 2, blue = 3). Both coefficients used defined white = 2, so when we `getUnion` on the attribute/index pairs we get the result we want.

The `set` infrastructure might be better extended to better allow for "remapping" of different attribute subsets. Currently I'm just copying dictionaries and list collections, but view semantics (or pairs of attributes - references) might be more efficient.